src/controllers/jobCoordinator.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  Controller,
+  Inject,
+  UseInterceptors,
+  ClassSerializerInterceptor,
+  Get,
+  Post,
+  Delete,
+  Param,
+  Body,
+  Put,
+  UseGuards,
+  HttpException,
+  HttpStatus,
+} from "@nestjs/common";
+import { AuthGuard } from "@nestjs/passport";
+import { ApiBearerAuth } from "@nestjs/swagger";
+import { JOB_COORDINATOR_SERVICE } from "src/constants";
+import { JobIdParamDto } from "src/dtos/job";
+import JobCoordinatorService from "src/services/JobCoordinatorService";
+import { JobCoordinator } from "src/entities/JobCoordinator";
+import { CreateJobCoordinatorsDto, JobCoordinatorIdParamDto,  UpdateJobCoordinatorDto } from "src/dtos/jobCoordinator";
+import { TransactionInterceptor } from "src/interceptor/TransactionInterceptor";
+import { TransactionParam } from "src/decorators/TransactionParam";
+import { Transaction } from "sequelize";
+import { UpdateOrFind } from "src/utils/utils";
+
+@Controller("/job/:jobId/coordinators")
+@ApiBearerAuth("jwt")
+@UseGuards(AuthGuard("jwt"))
+export class JobCoordinatorController {
+  constructor(@Inject(JOB_COORDINATOR_SERVICE) private jobCoordinatorService: JobCoordinatorService) {}
+
+  @Get()
+  @UseInterceptors(ClassSerializerInterceptor)
+  async getJobCoordinators(@Param() param: JobIdParamDto) {
+    const jobCoordinators = await this.jobCoordinatorService.getJobCoordinators({ jobId: param.jobId });
+    return { jobCoordinators: jobCoordinators };
+  }
+
+  @Post()
+  @UseInterceptors(ClassSerializerInterceptor)
+  @UseInterceptors(TransactionInterceptor)
+  async createJobCoordinators(@Param() param: JobIdParamDto, @Body() body: CreateJobCoordinatorsDto, @TransactionParam() transaction: Transaction) {
+    const promises = [];
+    for (const jobCoordinator of body.jobCoordinators) {
+      promises.push(
+        new Promise(async (resolve, reject) => {
+          try {
+            const newJobCoordinator = await this.jobCoordinatorService.createOrGetJobCoordinator(
+              new JobCoordinator({
+                jobId: param.jobId,
+                tpcMemberId: jobCoordinator.tpcMemberId,
+                role: jobCoordinator.role
+              }),
+              transaction
+            );
+            resolve(newJobCoordinator);
+          } catch (err) {
+            reject(err);
+          }
+        })
+      );
```

Comment:
```suggestion
      promises.push(this.jobCoordinatorService.createOrGetJobCoordinator(
              new JobCoordinator({
                jobId: param.jobId,
                tpcMemberId: jobCoordinator.tpcMemberId,
                role: jobCoordinator.role
              }),
              transaction
            ));
```

Reply:
We needed to create extra Promise object earlier because there were 2 await functions, here we have only one function and this function itself is a Promise